crates/iota-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -1581,4 +1637,217 @@ mod object_cost_tests {
+            for mutable_1 in [true, false].iter() {
+                let tx = build_transaction(
+                    &[(shared_obj_0, *mutable_0), (shared_obj_1, *mutable_1)],
+                    tx_gas_budget,
+                    1,
+                );
+                if let SequencingResult::Defer(_, congested_objects) =
+                    shared_object_congestion_tracker.try_schedule(
+                        &tx,
+                        max_execution_duration_per_commit,
+                        max_overshoot_per_commit,
+                        &HashMap::new(),
+                        0,
+                    )
+                {
+                    if assign_min_free_execution_slot {
+                        assert_eq!(congested_objects.len(), 2);
+                    } else {
+                        assert_eq!(congested_objects.len(), 1);
+                        assert_eq!(congested_objects[0], shared_obj_0);
+                    }
+                } else {
+                    panic!("should defer");
+                }
+            }
+        }
+    }
+
+    #[rstest]
+    fn test_accumulated_debts(
+        #[values(
+            PerObjectCongestionControlMode::TotalGasBudget,
+            PerObjectCongestionControlMode::TotalTxCount
+        )]
+        mode: PerObjectCongestionControlMode,
+        #[values(true, false)] assign_min_free_execution_slot: bool,
+    ) {
+        // Creates two shared objects and three transactions that operate on these
+        // objects.
+        let shared_obj_0 = ObjectID::random();
+        let shared_obj_1 = ObjectID::random();
+
+        let tx_gas_budget = 100;
+
+        // Set max_accumulated_txn_cost_per_object_in_commit  and initial_object_debt
+        // such that a single transaction will cause an overshoot.
+        let max_execution_duration_per_commit = match mode {
+            PerObjectCongestionControlMode::None => unreachable!(),
+            PerObjectCongestionControlMode::TotalGasBudget => 90,
+            PerObjectCongestionControlMode::TotalTxCount => 2,
+        };
+
+        let initial_object_debt = match mode {
+            PerObjectCongestionControlMode::None => unreachable!(),
+            PerObjectCongestionControlMode::TotalGasBudget => 80,
+            PerObjectCongestionControlMode::TotalTxCount => 2,
+        };
+
+        let mut shared_object_congestion_tracker = SharedObjectCongestionTracker::new_for_test(
+            [
+                (shared_obj_0, initial_object_debt),
+                (shared_obj_1, initial_object_debt),
+            ],
+            mode,
+            assign_min_free_execution_slot,
+        );
+
+        // Verify that accumulated_debts is empty initially.
+        let accumulated_debts = shared_object_congestion_tracker
+            .clone()
+            .accumulated_debts(max_execution_duration_per_commit);
+        assert!(accumulated_debts.is_empty());
+
+        // Simulate a tx on object 0 that exceeds the per-commit limit.
```

Comment:
```suggestion
        // Simulate a tx on object 0 that exceeds the per-commit limit,
        // taking into account the initial object debt.
```
In tx count mode, a tx would not exceed the per-commit limit, which is set to `2`, if the initial object debt was lower. In gas budget mode, a tx always exceeds the limit (`90`) as tx gas budget is set to `100`, no matter if there are initial objects debts or not.